compiler/rustc_typeck/src/check/wfcheck.rs

```diff
@@ -201,6 +201,40 @@ pub fn check_trait_item(tcx: TyCtxt<'_>, def_id: LocalDefId) {
     };
     check_object_unsafe_self_trait_by_name(tcx, &trait_item);
     check_associated_item(tcx, trait_item.hir_id(), span, method_sig);
+
+    let encl_trait_hir_id = tcx.hir().get_parent_item(hir_id);
+    let encl_trait = tcx.hir().expect_item(encl_trait_hir_id);
```

Comment:
I wonder if this will have a perf hit 🤔

If it does, we might be able to pass in the `encl_trait.def_id` from wherever `check_trait_item` is called (we might have already queried for it).